core/src/main/java/com/klaviyo/coresdk/networking/NetworkRequest.kt

```diff
@@ -0,0 +1,95 @@
+package com.klaviyo.coresdk.networking
+
+import android.content.Context
+import android.net.ConnectivityManager
+import android.net.NetworkCapabilities
+import android.os.Build
+import android.util.Base64
+import com.klaviyo.coresdk.KlaviyoConfig
+import java.io.*
+import java.net.HttpURLConnection
+import java.net.URL
+import javax.net.ssl.HttpsURLConnection
+
+
+abstract class NetworkRequest {
+    internal abstract var url: URL
+    internal abstract var requestMethod: RequestMethod
+    internal abstract var headerData: String?
+    internal abstract var payload: String?
+
+    internal fun isInternetConnected(context: Context): Boolean {
+        val connectivityManager = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
+        // TODO: We may want to replace this later. Deprecated as of Android API 29.
+        //  But the alternative solution is an asynchronous task that requires the user to register
+        //  a network callback listener, which isn't ideal just for a network connectivity check in an SDK
+        val networkInfo = connectivityManager.activeNetworkInfo ?: return false
```

Comment:
As noted in the comment above, this `NetworkInfo` class that we are fetching here is deprecated as of Android 29 sadly. And the alternative workaround is not as easy from an SDK-development standpoint. I vote we leave this in as a connectivity check for now but cut a ticket to consider moving to a network listener approach in the future.